test/functional/zmq_test.py

```diff
@@ -33,121 +57,68 @@ def setup_nodes(self):
         if not config["components"].getboolean("ENABLE_ZMQ"):
             raise SkipTest("bitcoind has not been built with zmq enabled.")
 
-        self.zmqContext = zmq.Context()
-        self.zmqSubSocket = self.zmqContext.socket(zmq.SUB)
-        self.zmqSubSocket.set(zmq.RCVTIMEO, 60000)
-        self.zmqSubSocket.setsockopt(zmq.SUBSCRIBE, b"hashblock")
-        self.zmqSubSocket.setsockopt(zmq.SUBSCRIBE, b"hashtx")
-        self.zmqSubSocket.setsockopt(zmq.SUBSCRIBE, b"rawblock")
-        self.zmqSubSocket.setsockopt(zmq.SUBSCRIBE, b"rawtx")
-        ip_address = "tcp://127.0.0.1:28332"
-        self.zmqSubSocket.connect(ip_address)
-        self.extra_args = [['-zmqpubhashblock=%s' % ip_address, '-zmqpubhashtx=%s' % ip_address,
-                       '-zmqpubrawblock=%s' % ip_address, '-zmqpubrawtx=%s' % ip_address], []]
+        # Initialize ZMQ context and socket.
+        # At the moment all messages are received in the same socket which means
+        # that this test fails if the publishing order changes.
+        # Note that the publishing order is not defined in the documentation and
+        # is subject to change.
+        self.zmq_context = zmq.Context()
+        self.socket = self.zmq_context.socket(zmq.SUB)
+        self.socket.set(zmq.RCVTIMEO, 60000)
+        self.socket.connect(self.address)
+
+        # Subscribe to all available topics.
+        self.subscribe("hashblock")
+        self.subscribe("hashtx")
+        self.subscribe("rawblock")
+        self.subscribe("rawtx")
```

Comment:
I found 'subscribe' a bit misleading here, though it could be that I'm thinking too much.
It's not merely a standard ZeroMQ subscription; the ZMQSubscriber class created tracks sequence. As it's written, my immediate thought was 'why aren't we just doing `self.subscribe("")`' (pre this commit, the four setsockopt calls could have been replaced with just one).
Perhaps 'create_subscriber' or fluffing up the comment a bit.

Reply:
I can rename.